### ISSUE TITLE
AtMetaData.fromJson incorrectly converting null to zero for ttr, ttl, ttb

### ISSUE THREAD

**Issue by gkc** [atsign-foundation/at_server#1210]
### Describe the bug

In [this code](https://github.com/atsign-foundation/at_server/blob/trunk/packages/at_persistence_secondary_server/lib/src/model/at_meta_data.dart#L116-L130) in at_meta_data.dart, values for ttl, ttb and ttr are being set to zero if the json value was null.
```dart
      ttr = (json[AT_TTR] is String)
          ? int.parse(json[AT_TTR])
          : (json[AT_TTR] == null)
              ? 0
              : json[AT_TTR];
```

### Steps to reproduce

* As `@alice`
  * create a  public record e.g. `public:foo.bar@alice` without setting ttl, ttb or ttr in metadata: `update:public:foo.bar@alice Testing`
  * do an `llookup:all:public:foo.bar@alice` - can see that the ttr is null, as expected
  ```
    @alice@ update:public:foo.bar@alice Testing
      => data:5
    @alice@ scan
      => data:                
        ["@alice:signing_privatekey@alice","public:foo.bar@alice","public:publickey@alice",
        "public:signing_publickey@alice"]
    @alice@ llookup:all:public:foo.bar@alice
      => data:{"key":"public:foo.bar@alice","data":"Testing","metaData":{"createdBy":"@alice",
        "updatedBy":"@alice", "createdAt":"2023-02-19 10:22:20.887Z",
        "updatedAt":"2023-02-19 10:22:20.887Z", "availableAt":"2023-02-19 10:22:20.887Z",
        "expiresAt":null,"refreshAt":null,"status":"active","version":0,"ttl":0,"ttb":0,
    --> "ttr":null,"ccd":null,"isBinary":false,"isEncrypted":false,"dataSignature":null,
        "sharedKeyEnc":null,"pubKeyCS":null,"encoding":null}}
  ```
* As `@bob`
  * do a `plookup:all:foo.bar@alice` - the ttr is correctly null, as the data received from `@alice` by `@bob` server is passed directly back to `@bob` client
    * The plookup verb handler creates a cached copy without checking the value of ttr. (It should only be caching when the value is not null, and not zero.)
  * now look up the cached copy: `llookup:all:cached:public:foo.bar@alice` - the ttr is now, incorrectly, zero
  ```
    @bob@ scan
      => data:["@bob:signing_privatekey@bob","public:publickey@bob","public:signing_publickey@bob"]
    @bob@ plookup:all:foo.bar@alice       
      => data:{"key":"public:foo.bar@alice","data":"Testing","metaData":{"createdBy":"@alice",
        "updatedBy":"@alice","createdAt":"2023-02-19 10:22:20.887Z",
        "updatedAt":"2023-02-19 10:22:20.887Z",
        "availableAt":"2023-02-19 10:22:20.887Z",
        "expiresAt":null,"refreshAt":null,"status":"active","version":0,"ttl":0,"ttb":0,
    --> "ttr":null,"ccd":null,"isBinary":false,"isEncrypted":false,"dataSignature":null,
        "sharedKeyEnc":null,"pubKeyCS":null,"encoding":null}}
    @bob@ llookup:all:cached:public:foo.bar@alice
      => data:{"key":"cached:public:foo.bar@alice","data":"Testing","metaData":{"createdBy":"@alice",
        "updatedBy":"@bob","createdAt":"2023-02-19 10:23:03.337Z",
        "updatedAt":"2023-02-19 10:23:03.337Z","availableAt":"2023-02-19 10:23:03.337Z",
        "expiresAt":null,"refreshAt":null,"status":"active","version":0,"ttl":0,"ttb":0,
    --> "ttr":0,"ccd":null,"isBinary":false,"isEncrypted":false,"dataSignature":null,
        "sharedKeyEnc":null,"pubKeyCS":null,"encoding":null}}
  ```

Note in the above that the `refreshAt` value is null. This is correct - a null `ttr` should result in a null `refreshAt`. However all of this together causes [this other bug](https://github.com/atsign-foundation/at_server/issues/1211).

### Expected behavior

* Do not transform nulls to zeroes for ttr, ttl or ttb

**Comment by gkc** [comment]
PS the fix will be in a PR later today

**Comment by gkc** [comment]
Fixed by #1239
